File: backend/GameVault.Api/Models/Game.cs
namespace GameVault.Api.Models;

/* Database data model for a game entry
 *
 * fields:
 * Id:              Primary key - used for Postgres to identify
 * Title:           Title of the game
 * Platform:        Hardware Platform the Game was built for (i.e.: PS1)
 * Release Date:    Date Game released
 * Publisher:       Game's Publisher
 * Developer:       Game's Developer
 * Description:     Brief Description of the game.
 * CoverArtUrl:     URL path to Art asset for game.
 * Region:          Region of Release
 *
 */
public class Game
{
    public int Id { get; set;}
    public string Title { get; set; } = string.Empty;
    public string Platform { get; set; } = string.Empty;
    public DateOnly? ReleaseDate { get; set; }
    public string? Publisher { get; set; }
    public string? Developer { get; set; }
    public string? Description { get; set; }
    public string? CoverArtUrl { get; set; }
    public string Region { get; set; } = string.Empty;
}

File: frontend/game-vault/src/types/game.ts
// Data format for receiving data from C# backend

export interface Game {
    id: number
    title: string
    platform: string
    releaseDate: string | null
    publisher: string | null
    developer: string | null
    description: string | null
    coverArtUrl: string | null
    region: string
}

export interface PaginatedResponse<T> {
    items: T[]
    totalCount: number
    page: number
    pageSize: number
}

export interface Platform {
    platform: string
    game_count: number
}
